app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelDialogFragment.kt

```diff
@@ -172,12 +170,24 @@ class TrackingProtectionPanelDialogFragment : AppCompatDialogFragment(), BackHan
 
     private fun toggleTrackingProtection(isEnabled: Boolean) {
         context?.let { context ->
-            val host = url.tryGetHostFromUrl()
-            lifecycleScope.launch {
-                ExceptionDomains(context).toggle(host)
-            }
-            with(context.components) {
-                useCases.sessionUseCases.reload.invoke(core.sessionManager.findSessionById(sessionId))
+            val useCase = TrackingProtectionUseCases(
+                sessionManager = context.components.core.sessionManager,
+                engine = context.components.core.engine
+            )
+            val session = context.components.core.sessionManager.findSessionById(sessionId)
+            session?.let {
+                useCase.containsException(it) { contains ->
```

Comment:
I'm not sure why we are checking if the session is on the list. I thought we can relay on `isEnabled` to determine which action to take. Is there any particular reason for checking?

Reply:
Nope you're right :)